docs/source/checkpointing.rst

```diff
@@ -230,24 +230,99 @@ with `idx` parameter always unset, and the same :class:`~.Function` can only be
 loaded using the same mode.
 
 
-Using disk checkpointing in adjoint simulations
+Using checkpointing in adjoint simulations
 ===============================================
 
 When adjoint annotation is active, the result of every Firedrake operation is
-stored in memory. For some simulations, this can result in a very large memory
-footprint. As an alternative, it is possible to specify that those intermediate
-results in forward evaluations of the tape which have type
-:class:`~firedrake.function.Function` be written to disk. This is usually the
-bulk of the data stored on the tape so this largely alleviates the memory
-problem, at the cost of the time taken to read to and write from disk.
+stored in memory. For some time-dependent simulations, this can lead to a
+large memory footprint. To alleviate this, checkpointing strategies can be used
+to store intermediate results either in memory or on disk.
 
-Having imported `firedrake.adjoint`, there are two steps required to enable
-disk checkpointing of the forward tape state.
+Checkpointing for time-dependent adjoint simulations in Firedrake employs
+schedules, which determine how forward data is stored. These schedules are
+implemented in the `checkpoint_schedules package
+<https://www.firedrakeproject.org/checkpoint_schedules/>`_.
 
-1. Call :func:`~firedrake.adjoint_utils.checkpointing.enable_disk_checkpointing`.
-2. Wrap all mesh constructors in :func:`~firedrake.adjoint_utils.checkpointing.checkpointable_mesh`.
 
-See the documentation of those functions for more detail.
+To store every time step of the forward data required for adjoint-based gradient
+computation **in memory**, use the following code:
+
+.. code-block:: python3
+
+    from firedrake import *
+    from firedrake.adjoint import *
+    from checkpoint_schedules import SingleMemoryStorageSchedule
+
+    continue_annotation()
+    tape = get_working_tape()
+    tape.enable_checkpointing(SingleMemoryStorageSchedule())
+
+**For any checkpointing approach, it is essential to call the time loop as
+follows when advancing the solver in time:**
+
+.. code-block:: python3
+
+    for step in tape.timestepper(range(total_steps)):
+        # Advance the forward model
+        # ...
+
+
+`SingleMemoryStorageSchedule`` stores only the last time step of the gradient in memory.
```

Comment:
```suggestion
``SingleMemoryStorageSchedule`` stores only the last time step of the gradient in memory.
```

Reply:
Let's be careful about how we describe what is stored. Maybe talk about which adjoint variables are retained.